.github/workflows/docs.yaml

```diff
@@ -0,0 +1,50 @@
+name: Docs
+
+on:
+    push:
+        branches:
+            - main
+
+jobs:
+    docs:
+        runs-on: ubuntu-latest
+
+        steps:
+        - uses: actions/checkout@v2
+
+        - name: Setup Python
+          uses: actions/setup-python@v2
+          with:
+              python-version: 3.8
+
+        - name: Install Dependencies
+          run: pip install .[dev]
+
+        - name: Build HTML artifact
+          run: "sphinx-build docs docs/_build"
+
+
+        - name: Upload HTML artifact
+          uses: actions/upload-artifact@v1
+          with:
+              name: DocumentationHTML
+              path: docs/_build/
+
+        # TODO: branch-specific folders for branches other than `main`
+        - name: Commit and publish documentation changes to gh-pages branch
+          run: |
+              git clone https://github.com/ApeWorX/eip712 --branch gh-pages --single-branch gh-pages
+              cp -r docs/_build/* gh-pages/
```

Comment:
Should this push to a `latest/` folder? (Then add `cp -r latest/ stable/` and `cp -r stable/ {{ TAG_NAME }}` to "on release" trigger)

Reply:
Yes I'll do that before merging this.

Reply:
This is now handled by the docs build script (the `python release_docs.py` command a few lines above).